src/test/java/com/google/devtools/build/lib/bazel/repository/posix/LocalConfigPlatformFunctionPosixTest.java

```diff
@@ -0,0 +1,83 @@
+// Copyright 2018 The Bazel Authors. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.devtools.build.lib.bazel.repository;
+
+import static com.google.common.truth.Truth.assertThat;
+
+import com.google.common.collect.ImmutableList;
+import com.google.devtools.build.lib.analysis.ConfiguredTarget;
+import com.google.devtools.build.lib.analysis.platform.ConstraintSettingInfo;
+import com.google.devtools.build.lib.analysis.platform.ConstraintValueInfo;
+import com.google.devtools.build.lib.analysis.platform.PlatformInfo;
+import com.google.devtools.build.lib.analysis.platform.PlatformProviderUtils;
+import com.google.devtools.build.lib.analysis.util.BuildViewTestCase;
+import com.google.devtools.build.lib.cmdline.Label;
+import com.google.devtools.build.lib.util.CPU;
+import com.google.devtools.build.lib.util.OS;
+import java.util.Collection;
+import org.junit.Test;
+import org.junit.experimental.runners.Enclosed;
+import org.junit.runner.RunWith;
+import org.junit.runners.JUnit4;
+import org.junit.runners.Parameterized;
+import org.junit.runners.Parameterized.Parameters;
+
+/** Posix-only tests for {@link LocalConfigPlatformFunction}. */
+@RunWith(Enclosed.class)
+public class LocalConfigPlatformFunctionPosixTest {
```

Comment:
What is it about this test that is posix-specific? Something about using CPU or OS?

Reply:
Running this test on Windows fails in this line:

https://github.com/bazelbuild/bazel/blob/1c082157ea1163ae3a6b858eb02aa290581879a8/src/test/java/com/google/devtools/build/lib/bazel/repository/posix/LocalConfigPlatformFunctionPosixTest.java#L66

with this message:
```
1) generateConfigRepository(com.google.devtools.build.lib.bazel.repository.LocalConfigPlatformFunctionPosixTest$FunctionTest)
java.lang.AssertionError: ERROR /usr/local/google/_blaze_jrluser/FAKEMD5/external/local_config_platform_test/BUILD.bazel:3:1: no such package '@bazel_tools//platforms': /usr/local/google/_blaze_jrluser/FAKEMD5/external/bazel_tools must be an existing directory and referenced by '@local_config_platform_test//:host'
```

Reply:
I see, this is a problem in the test setup, not a fundamental limitation of this class's windows support.

If I fix the test setup, would this class be moved back to the main repository tests package and re-named?

Reply:
Filed #7079 for myself to get this fixed.

Reply:
Sure, I'm happy to maximize test coverage on Windows! Thanks!